packages/core/package.json

```diff
@@ -5,6 +5,13 @@
   "main": "lib/index.js",
   "module": "es/index.js",
   "types": "lib/index.d.ts",
+  "typesVersions": {
+    "<4.1": {
+      "lib/types.4.1.d.ts": [
+        "types.pre-4.1.d.ts"
+      ]
+    }
+  },
```

Comment:
I still need to verify in some external project that this will be resolved correctly